qemu/softmmu/memory.c

```diff
@@ -148,6 +193,21 @@ MemoryRegion *memory_map_io(struct uc_struct *uc, ram_addr_t begin, size_t size,
     return mmio;
 }
 
+void memory_region_filter_subregions(MemoryRegion *mr, int32_t level)
+{
+    MemoryRegion *subregion, *subregion_next;
+    memory_region_transaction_begin();
+    QTAILQ_FOREACH_SAFE(subregion, &mr->subregions, subregions_link, subregion_next) {
+        if (subregion->priority >= level) {
+            memory_region_del_subregion(mr, subregion);
+            subregion->destructor(subregion);
+            g_free(subregion);
+            mr->uc->memory_region_update_pending = true;
```

Comment:
This seems not necessary? Though it is harmless.

Reply:
Not sure what you referencing but I would guess the `update_padding`:

This will force the `FlatView` to be rebuild. Without this later access to memory might go to removed regions. This could also done by rebuild the `FlatView` only after all `MemoryRegion` are filtered or by rebuild the `FlatView` forced after a `snapshot_restore`